src/spring-cloud/azext_spring_cloud/custom.py

```diff
@@ -101,9 +105,23 @@ def spring_cloud_create(cmd, client, resource_group, name, location=None,
     _update_application_insights_asc_create(cmd, resource_group, name, location,
                                             app_insights_key, app_insights, sampling_rate,
                                             disable_app_insights, no_wait)
+    if sku.name == 'E0':
```

Comment:
Moving this to validator, if the sku != enterprise, the `build_pool_size` should be None here, then we can skip the update on the param is None